Add yaml support.

EvgeneOskin opened the pull request.

sjaensch commented on swagger_spec_validator/common.py:
```diff
@@ -24,8 +25,16 @@ def wrapper(*args, **kwargs):
+
+
+def load_yaml(url):
+    _load(url, yaml.loads)
```
This should be yaml.safe_load if I'm not mistaken.

sjaensch commented:
Hi @EvgeneOskin, thank you for your contribution! Currently the build is red, will you be fixing it? That would be awesome.
Also, there might be issues with specs in the yaml format that don't quote the response status code (#49). You might want to add a workaround similar to what was done for bravado.

EvgeneOskin commented:
Hi @sjaensch, thank you for review and notes, I've fixed the build. Add simple test to load yml and json data. Also, I've fixed paths in `validator12.run_test`. At the moment, the build fails only in doc environment.

dimbleby commented on swagger_spec_validator/util.py:
```diff
@@ -48,6 +49,12 @@ def validate_spec_url(spec_url):
     """
-    spec_json = load_json(spec_url)
+    _, extension = os.path.splitext(spec_url)
+    is_yaml = any([i == extension for i in ('.yml', '.yaml')])
```
why not:
```python
is_yaml = extension in ['.yml', '.yaml']
```
?

EvgeneOskin replied:
You're right. I've improved it.

larsbutler commented:
@EvgeneOskin @sjaensch This branch appears to just need a rebase on the latest master, due to https://github.com/Yelp/swagger_spec_validator/commit/a4669b8dd05391fdb528d90af974b7ee7a003f6f. After that, it should work.

EvgeneOskin commented:
Thank @larsbutler

larsbutler commented:
@EvgeneOskin @sjaensch Hmmm, something seems to be stuck with coveralls. The check is still pending. It'd be great to get this patch in; I would use it right away. :)
